### PR TITLE
feat: improved empty screens

### LINKED ISSUE #3602: Filtered container list gives "no containers" message which is a bit of a lie
### Bug description

I was working on a wordpress container. I have a lot of containers in my container list, as you can see:

Screenshot from 2023-08-19 22-34-13

So I typed "wordpress" in the filter box at the top. And I deleted that wordpress container (going to recreate it with different options, I am debugging something.) Anyway, whoops, I had a minor panic attack after I deleted the wordpress container, because it now says I have no containers on my system:

Screenshot from 2023-08-19 22-32-59

But this is a lie :) I only have no containers that match the filter string "wordpress". 

## Suggestion

Add some logic to the empty list message, that if a filter is in place, the user can remove the filter to see their containers (rather than telling them to start a new throwaway example container.)

Here's an idea for how that could look:

Screenshot from 2023-08-19 22-48-17

Maybe this has a scaling feature though if the user types something reallysupercajufrajilistickexpialladocious... in which case maybe something like:

```
No containers matching filter found

No containers matched "reallysupercajufrajilistickexpialladocious"
Not what you expected? Double-check your spelling.
Just want to view all of your containers?
[ Clear filter ]
```

### Operating system

Fedora 38 (fedora-release-common-38-36.noarch) // Podman 4.6.1

### Installation Method

Flathub (Linux)

### Version

1.3.1

### Steps to reproduce

See steps above.

### Relevant log output

```shell
N/A
```

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by deboer-tim**
### What does this PR do?

Improves the empty-screen message when the cause is a filter that doesn't match anything.

The design requires adding second line in EmptyScreen, which I've done as a detail message. I then looked at making a *FilterEmptyScreen for each page as it's own component or inline, but there was a lot of copy/paste so I created a generic FilteredEmptyScreen. As per the design in issue #3602, when the filter is too long (I picked 20 chars) the search term isn't shown.

searchTerm was already exported from ContainerList, I exported it from the others as well since it is necessary to test, and a useful feature in general. Tests added to show the filter empty screen comes up when the filter doesn't match anything.

### Screenshot/screencast of this PR

https://github.com/containers/podman-desktop/assets/19958075/13e724f7-252f-4915-bb04-00665001d21d

### What issues does this PR fix or reference?

Fixes #3602.

### How to test this PR?

Type a filter that matches (to confirm no regression), doesn't match (to confirm new behaviour), and is really long (to show the message changes).

### CODE REVIEW

File: packages/renderer/src/lib/ContainerList.svelte [hunk 1]
```diff
@@ -641,7 +643,11 @@ function errorCallback(container: ContainerInfoUI, errorMessage: string): void {
     {#if providerConnections.length === 0}
       <NoContainerEngineEmptyScreen />
     {:else if $filtered.length === 0}
-      <ContainerEmptyScreen />
+      {#if searchTerm.length > 0}
```
Comment (benoitf) [inline packages/renderer/src/lib/ContainerList.svelte]: you can simplify with

```suggestion
      {#if searchTerm}
```

File: packages/renderer/src/lib/ImagesList.svelte [hunk 2]
```diff
@@ -328,7 +329,11 @@ function computeInterval(): number {
     {#if providerConnections.length === 0}
       <NoContainerEngineEmptyScreen />
     {:else if $filtered.length === 0}
-      <ImageEmptyScreen />
+      {#if searchTerm.length > 0}
```
Comment (benoitf) [inline packages/renderer/src/lib/ImagesList.svelte]:
```suggestion
      {#if searchTerm}
```

File: packages/renderer/src/lib/pod/PodsList.svelte [hunk 3]
```diff
@@ -306,7 +307,11 @@ function errorCallback(pod: PodInfoUI, errorMessage: string): void {
     {#if $filtered.length === 0 && providerConnections.length === 0}
       <NoContainerEngineEmptyScreen />
     {:else if $filtered.length === 0}
-      <PodEmptyScreen />
+      {#if searchTerm.length > 0}
```
Comment (benoitf) [inline packages/renderer/src/lib/pod/PodsList.svelte]:
```suggestion
      {#if searchTerm}
```

File: packages/renderer/src/lib/volume/VolumesList.svelte [hunk 4]
```diff
@@ -288,7 +289,11 @@ function gotoCreateVolume(): void {
     {#if providerConnections.length === 0}
       <NoContainerEngineEmptyScreen />
     {:else if $filtered.map(volumeInfo => volumeInfo.Volumes).flat().length === 0}
-      <VolumeEmptyScreen />
+      {#if searchTerm.length > 0}
```
Comment (benoitf) [inline packages/renderer/src/lib/volume/VolumesList.svelte]:
```suggestion
      {#if searchTerm}
```

**Review by benoitf — APPROVED** [review]

**Review by lstocchi — APPROVED** [review]
LGTM

File: packages/renderer/src/lib/volume/VolumesList.svelte (hunk 4 above)
Comment (deboer-tim) [reply packages/renderer/src/lib/volume/VolumesList.svelte]: Fixed.

File: packages/renderer/src/lib/pod/PodsList.svelte (hunk 3 above)
Comment (deboer-tim) [reply packages/renderer/src/lib/pod/PodsList.svelte]: Fixed.

File: packages/renderer/src/lib/ContainerList.svelte (hunk 1 above)
Comment (deboer-tim) [reply packages/renderer/src/lib/ContainerList.svelte]: Fixed.

File: packages/renderer/src/lib/ImagesList.svelte (hunk 2 above)
Comment (deboer-tim) [reply packages/renderer/src/lib/ImagesList.svelte]: Fixed.

File: packages/renderer/src/lib/ui/FilteredEmptyScreen.svelte [hunk 5]
```diff
@@ -0,0 +1,23 @@
+<script lang="ts">
+import EmptyScreen from '../ui/EmptyScreen.svelte';
+import Button from '../ui/Button.svelte';
+
+export let icon: any;
+export let kind: string;
+export let searchTerm: string;
+
+$: filter = searchTerm && searchTerm.length > 20 ? 'filter' : `'` + searchTerm + `'`;
```
Comment (benoitf) [inline packages/renderer/src/lib/ui/FilteredEmptyScreen.svelte]: you should use
```
 `'${searchTerm}'`
```

File: packages/renderer/src/lib/ui/EmptyScreen.svelte [hunk 6]
```diff
@@ -6,6 +6,7 @@ import { onMount } from 'svelte';
 export let icon: any;
 export let title = 'No title';
 export let message = 'Message';
+export let detail = undefined;
```
Comment (benoitf) [inline packages/renderer/src/lib/ui/EmptyScreen.svelte]: need to init to empty string probably so we know it's a string type

File: packages/renderer/src/lib/ui/FilteredEmptyScreen.svelte (hunk 5 above)
Comment (deboer-tim) [reply packages/renderer/src/lib/ui/FilteredEmptyScreen.svelte]: Fixed.

File: packages/renderer/src/lib/ui/EmptyScreen.svelte (hunk 6 above)
Comment (deboer-tim) [reply packages/renderer/src/lib/ui/EmptyScreen.svelte]: Done.

**Review by benoitf — APPROVED** [review]

**Pull request merged**